src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Design.Serialization;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms;
+
+[Export(typeof(IMyAppFileAccessor))]
+[AppliesTo(ProjectCapability.DotNet)]
+internal class MyAppFileAccessor : IMyAppFileAccessor, IDisposable
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly UnconfiguredProject _project;
+    private readonly IProjectThreadingService _threadingService;
+    private DocData? _docData;
+    private MyAppDocument? myAppDocument;
+
+    private const string MySubMainProperty = "MySubMain";
+    private const string MainFormProperty = "MainForm";
+    private const string SingleInstanceProperty = "SingleInstance";
+    private const string ShutdownModeProperty = "ShutdownMode";
+    private const string EnableVisualStylesProperty = "EnableVisualStyles";
+    private const string AuthenticationModeProperty = "AuthenticationMode";
+    private const string SaveMySettingsOnExitProperty = "SaveMySettingsOnExit";
+    private const string HighDpiModeProperty = "HighDpiMode";
+    private const string SplashScreenProperty = "SplashScreen";
+    private const string MinimumSplashScreenDisplayTimeProperty = "MinimumSplashScreenDisplayTime";
+
+    [ImportingConstructor]
+    public MyAppFileAccessor([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, UnconfiguredProject project, IProjectThreadingService threadingService)
+    {
+        _serviceProvider = serviceProvider;
+        _project = project;
+        _threadingService = threadingService;
+    }
+
+    public void Dispose()
+    {
+        _docData?.Dispose();
+    }
+
+    private void DocData_Modifying(object sender, EventArgs e)
+    {
+        myAppDocument = null;
+    }
+
+    private async Task<MyAppDocument?> TryGetMyAppFileAsync()
+    {
+        if (_docData is null)
+        {
+            string absolutePath = _project.MakeRooted("My Project\\Application.myapp");
+
+            await _threadingService.SwitchToUIThread();
+            _docData = new DocData(_serviceProvider, absolutePath);
+            _docData.Modifying += DocData_Modifying;
+            await TaskScheduler.Default;
+        }
+
+        if (myAppDocument is null)
+        {
+            myAppDocument = new MyAppDocument(_docData);
+        }
+
+        return myAppDocument;
+
+        //return null; // Consider when we will reach here: when file doesn't exist.
```

Comment:
You should decide one way or another about this before merging.

Reply:
Decided to do a try/catch when creating the DocData, and return null when we can't create it.